.github/workflows/unit_test.yml

```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Install Python + Retrieve Poetry dependencies from cache
         uses: actions/setup-python@v4
         with:
-          python-version: '3.9'
+          python-version: ${{ env.PYTHON_VERSION }}
```

Comment:
Why was it default to 3.9 before? / What is this doing?

Reply:
If you look higher up in the script, you'll see that `PYTHON_VERSION` is set as an env var and is set to 3.10. I figured the version that linting was using was supposed to be using 3.10. Also, IMO it's better for the linter to use the most modern fully-supported version of Python, because it'll likely have the best performance.